### PR TITLE
Fixed ASGenerator.GetStartOfStatement for array comprehension. fixes #1883

### LINKED ISSUE #1883: [Bug][ASgenerator][GetStartOfStatement] Wrong result for array comprehension
```haxe
 [1 => 1, 2 => 2]|
```
`|` - cursor
expected result: 1
actual result: 0

### PR DESCRIPTION
**Pull request by SlavaRa**

### CODE REVIEW

File: External/Plugins/ASCompletion/Completion/ASGenerator.cs [hunk 1]
```diff
@@ -1982,11 +1982,11 @@ public static int GetStartOfStatement(ScintillaControl sci, int statementEnd, AS
                 else if (c == '[' && arrCount > 0) arrCount--;
                 else if (c == ')') parCount++;
                 else if (c == '(' && parCount > 0) parCount--;
-                else if (c == '>')
+                else if (c == '>' && arrCount == 0 && parCount == 0 && braCount == 0)
```
Comment (Neverbirth) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: I'm really curious if this is 100% fine, as there are many cases were generics may appear between []{}(), but I cannot check at the moment, also it seems there are no tests with this?. For example `arr = [new Array<String>()]`, `function test(type:Class<Dynamic>)`, etc.

Comment (SlavaRa) [reply External/Plugins/ASCompletion/Completion/ASGenerator.cs]: 5 min

Comment (SlavaRa) [reply External/Plugins/ASCompletion/Completion/ASGenerator.cs]: All for you ^__^

Comment (SlavaRa) [reply External/Plugins/ASCompletion/Completion/ASGenerator.cs]: generics is not a problem, problem is operators `<`, `>`

Comment (SlavaRa) [reply External/Plugins/ASCompletion/Completion/ASGenerator.cs]: >For example arr = [new Array<String>()], function test(type:Class<Dynamic>), etc.

2918: `new TestCaseData(" new Vector.<int>()$(EntryPoint)"`
2927: `new TestCaseData(" new <Vector.<Object>>[new <Object>[{}]]$(EntryPoint)"`
2970: `new TestCaseData(" new Array<{name:String, params:Array<Dynamic>}>()$(EntryPoint)"`

Comment (Neverbirth) [reply External/Plugins/ASCompletion/Completion/ASGenerator.cs]: I see, but in Haxe "=>" is an operator and could appear in other places to be later processed at macro time, so "=>" should be processed specifically and not assume it's going to be between brackets.

Comment (SlavaRa) [reply External/Plugins/ASCompletion/Completion/ASGenerator.cs]: in the next PRs there will be a lot of them

**Pull request merged**